### PR TITLE
Add coverage for GPU parts

### PR DESCRIPTION
**Pull request by abergeron**
Fixes a number of issues that were present in the GPU parts of the backends.

This also updates flake8 checking to use 3.7.7

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[flake8]
+ignore = W503,W504,E241,F722
```
Comment (breuleux) [inline .flake8]: GitHub reports no newline at end of file and it's bugging me.

File: myia/compile/backends/nnvm.py [hunk 2]
```diff
@@ -314,7 +314,7 @@ def convert(self, lst, context):
         target = context.MASK2STR[context.device_type]
         if target == 'cpu':
             nnvm_target = 'llvm'
-        elif target == 'cuda':  # pragma: no cover
+        elif target == 'gpu':
```
Comment (breuleux) [inline myia/compile/backends/nnvm.py]: Why are we using `'gpu'` instead of `'cuda'`?

File: tests/compile/test_nnvm.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from myia import dtype
+
+try:
+    from myia.compile.backends import nnvm
+except ImportError:
+    pytestmark = pytest.mark.skip(f"Can't import nnvm")
+
+
+@pytest.mark.gpu
+def test_nnvm_backend_exists():
```
Comment (breuleux) [inline tests/compile/test_nnvm.py]: I find the test name a tad misleading.

File: myia/compile/backends/nnvm.py (hunk 2 above)
Comment (abergeron) [reply myia/compile/backends/nnvm.py]: Because that's what tvm uses.

**Pull request merged**